### PR TITLE
[chore][CONTRIBUTING.md] Outline rules for moving contributors to emeritus

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** <Describe what has changed.>

Outlines rules for moving a contributor to emeritus from an existing role.

**Link to tracking Issue:** Relates to #34186

### CODE REVIEW

**Comment by mx-psi** [comment]
cc @open-telemetry/collector-contrib-triagers

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -426,6 +432,14 @@ To become a Code Owner, open a PR with the following changes:
 
 Be sure to tag the existing Code Owners, if any, within the PR to ensure they receive a notification.
 
+### Emeritus roles
+
+Contributors that are stepping down or reducing their role in the project are encouraged to move to [emeritus](https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#emeritus-maintainerapprovertriager). 
+
+In the event that a contributor steps down without prior notice, the maintainers will attempt to contact the contributor via both Github and the CNCF Slack to confirm their status. After two weeks, if the contributor is an approver or maintainer, they may be removed from the Github review auto-assignment. If the contributor does not respond within a period of one month, they may be moved to emeritus status at the discretion of the maintainers.
```
Comment (ChrsMark) [inline CONTRIBUTING.md:439]: Do we have clear expectations for a contributor's involvement?
What a contributor with a specific role should do in case there is planned "time off" for any kind of reasons?
Some examples can be when people go on long time off (1+ month), time off for medical reasons, paternity leaves etc.

Comment (mx-psi) [reply CONTRIBUTING.md:439]: > Not sure if that's on-topic but reading the "timeline" based steps I wonder if we have clear expectations for a contributor's involvement?

That's fair, yes, I think we should mention something related to this. The timeline is meant to apply in case there is no prior notice, maybe defining what that prior notice is would be enough? (e.g. a message/status on Slack is what people typically do, we could mention that explicitly).

> Some examples can be when people go on long time off (1+ month), time off for medical reasons, paternity leaves etc.

Personally I am fine with making the second part (moving to emeritus) longer, I think the more important bit is that the day-to-day activities continue, so I am fine with a short timeline for "remove from autoassignment" and a longer one for "move to emeritus"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -426,6 +432,14 @@ To become a Code Owner, open a PR with the following changes:
...
+
+In case of disagreement, this will be submitted to a majority vote among the maintainers (possibly excluding the contributor in question).
```
Comment (evan-bradley) [inline CONTRIBUTING.md:441]: Can we define who disagrees about what?

Comment (mx-psi) [reply CONTRIBUTING.md]: I am torn about this, maybe we should just say that we must make a majority vote, that would eliminate some ambiguity and possibility for bias

File: CONTRIBUTING.md (hunk 1 above)
Comment (mx-psi) [reply CONTRIBUTING.md]: PTAL at 20fd76856b1047b019407f903fdf35bc61716230

File: main.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+extensions:
```
Comment (songy23) [inline main.yaml:1]: Is this file relevant to this PR?

Comment (mx-psi) [reply main.yaml]: Oops, thanks for catching that

File: CONTRIBUTING.md (hunk 2 above)
Comment (mx-psi) [reply CONTRIBUTING.md]: Reworded as discussed in the SIG meeting

File: CONTRIBUTING.md (hunk 1 above)
Comment (mx-psi) [reply CONTRIBUTING.md]: Marking as solved as we discussed this on the SIG meeting

**Review by ChrsMark — APPROVED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -426,6 +432,14 @@ To become a Code Owner, open a PR with the following changes:
 
 Be sure to tag the existing Code Owners, if any, within the PR to ensure they receive a notification.
 
+### Emeritus roles
+
+Contributors that are stepping down or reducing their role in the project are encouraged to move to [emeritus](https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#emeritus-maintainerapprovertriager). In case of long temporary absences, contributors are encouraged to let maintainers know on the CNCF Slack #otel-collector-dev channel and to mark themselves as 'Busy' on Github.
+
+In the event that a contributor steps down without prior notice, the maintainers will attempt to contact the contributor via both Github and the CNCF Slack to confirm their status. After two weeks, if the contributor is an approver or maintainer, they may be removed from the Github review auto-assignment.
```
Comment (songy23) [inline CONTRIBUTING.md:439]:
```suggestion
In the event that a contributor becomes inactive without prior notice, the maintainers will attempt to contact the contributor via both Github and the CNCF Slack to confirm their status. After two weeks, if the contributor is an approver or maintainer, they may be removed from the Github review auto-assignment.
```

It sounds to me step down is the consequence

**Review by songy23 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -426,6 +432,14 @@ To become a Code Owner, open a PR with the following changes:
 
 Be sure to tag the existing Code Owners, if any, within the PR to ensure they receive a notification.
 
+### Emeritus roles
+
+Contributors that are stepping down or reducing their role in the project are encouraged to move to [emeritus](https://github.com/open-telemetry/community/blob/main/guides/contributor/membership.md#emeritus-maintainerapprovertriager). In case of long temporary absences, contributors are encouraged to let maintainers know on the CNCF Slack #otel-collector-dev channel and to mark themselves as 'Busy' on Github.
```
Comment (djaglowski) [inline CONTRIBUTING.md:437]: > Contributors that are stepping down or reducing their role

Maybe we should say something like "Contributors who are unable to meet the responsibilities of their role". The reason I suggest this is that "reducing their role" could be taken to mean a minor reduction in time spent, which occurs frequently in the normal course of operations for many people.

> on the CNCF Slack #otel-collector-dev channel

I would be in favor of removing this detail for privacy reasons. Maybe offer it as an option but sometimes prolonged absences would be better communicated privately. IMO it's enough that it is communicated to a maintainer, who can then handle further communication if necessary.

File: CONTRIBUTING.md (hunk 4 above)
Comment (djaglowski) [inline CONTRIBUTING.md:439]: I get what you're saying here but "steps down" and "without prior notice" are a bit contradictory to me because "stepping down" involves explicit communication. Maybe we could say "If two or more maintainers agree that a contributor does not appear to be fulfilling the responsibilities of their role"?

Comment (mx-psi) [reply CONTRIBUTING.md:439]: What do you think about 'becomes inactive' as Yang suggested?

Comment (codeboten) [reply CONTRIBUTING.md:439]: I support this edit

**Review by Frapschen — APPROVED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (djaglowski) [reply CONTRIBUTING.md]: I support it

**Review by atoulme — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Review by TylerHelmuth — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
Thanks @mx-psi

**Review by djaglowski — APPROVED** [review]
